python-sdk/tutorials/nuimages_tutorial.ipynb

```diff
@@ -0,0 +1,435 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sd_token_camera = sample['key_camera_token']\n",
+    "im = nuim.render_image(sd_token_camera)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Every annotated image (sample) comes with up to 6 past and 6 future images, spaced evenly at 500ms +- 250ms.  For each image we have a matching lidar pointcloud. However, a small percentage of the samples has less sample_datas, either because they were at the beginning or end of a log, or due to delays or dropped data packages.\n",
+    "`list_sample_content()` shows for each sample all the associated sample_datas, which are images (from the same camera) and lidar pointclouds."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nuim.list_sample_content(sample['token'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Besides the annotated images, we can also render the 6 previous and 6 future images, which are not annotated. Let's select the next image, which is taken around 0.5s after the annotated image. We can either manually copy the token from the list above or use the `next` pointer of the `sample_data`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "next_sd_token_camera = nuim.get('sample_data', sd_token_camera)['next']\n",
+    "next_sd_token_camera"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have the next token, let's render it. Note that we cannot render the annotations, as they don't exist.\n",
+    "\n",
+    "*Note: If you did not download the non-keyframes (sweeps), this will throw an error! We make sure to catch it here.*"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    im = nuim.render_image(next_sd_token_camera, with_annotations=False)\n",
+    "except Exception as e:\n",
+    "    print('As expected, we encountered this error:', e)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As mentioned, most images have an associated lidar pointcloud. We can project this pointcloud to the image and display it using `render_depth()`. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nuim.render_depth(sd_token_camera, mode='sparse')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since the pointcloud is relatively sparse, this image is hard to see. We can enable depth completion of the pointcloud to get a denser version of the image. To get the raw data of these plot, use `get_depth()`."
```

Comment:
To get the raw data of ~~these~~ this plot

Reply:
Also, this sentence talks about using `get_depth()`, but the example immediately following that is `render_depth()` instead. `get_depth()` is buried within `render_depth()` in the code so that makes sense after I have read the code, but would it be confusing to a first-time user? Maybe better to say this instead?:
```
To get the raw data of these plot, set `mode='dense`'
```